### PR TITLE
Add DID sign message API

### PR DESCRIPTION
**Pull request by ytx1991**
Follow the CHIP0002 standard

### CODE REVIEW

File: chia/wallet/did_wallet/did_wallet.py [hunk 1]
```diff
@@ -1095,6 +1096,26 @@ def get_parent_for_coin(self, coin) -> Optional[LineageProof]:
...
+            message = bytes.fromhex(hex_message)
+            p2_puzzle, _, _, _, _ = puzzle_args
+            puzzle_hash = p2_puzzle.get_tree_hash()
+            pubkey, private = await self.wallet_state_manager.get_keys(puzzle_hash)
+            synthetic_secret_key = calculate_synthetic_secret_key(private, DEFAULT_HIDDEN_PUZZLE_HASH)
+            synthetic_pk = synthetic_secret_key.get_g1()
+            prefix = f"\x18Chia Signed Message:\n{len(message)}"
+            return str(synthetic_pk), str(
+                AugSchemeMPL.sign(synthetic_secret_key, std_hash(prefix.encode("utf-8") + message))
+            )
```
Comment (Quexington) [inline chia/wallet/did_wallet/did_wallet.py:1115]: The wallet should be returning `Tuple[G1Element, G2Element]` and the RPC should convert to strings if it so desires.

File: chia/wallet/did_wallet/did_wallet.py [hunk 2]
```diff
@@ -1095,6 +1096,26 @@ def get_parent_for_coin(self, coin) -> Optional[LineageProof]:
 
         return parent_info
 
+    async def sign_message(self, hex_message: str) -> Tuple[str, str]:
```
Comment (Quexington) [inline chia/wallet/did_wallet/did_wallet.py:1099]: Take `bytes` as a message, not `str`

File: chia/wallet/did_wallet/did_wallet.py [hunk 3]
```diff
@@ -1095,6 +1096,26 @@ def get_parent_for_coin(self, coin) -> Optional[LineageProof]:
...
+        if self.did_info.current_inner is None:
+            raise ValueError("Missing DID inner puzzle.")
+        puzzle_args = did_wallet_puzzles.uncurry_innerpuz(self.did_info.current_inner)
+        if puzzle_args is not None:
+            if hex_message.startswith("0x") or hex_message.startswith("0X"):
+                hex_message = hex_message[2:]
```
Comment (Quexington) [inline chia/wallet/did_wallet/did_wallet.py:1105]: Use `bytes32.from_hexstr()`

**Review by Quexington — CHANGES_REQUESTED** [review]

File: chia/wallet/did_wallet/did_wallet.py (hunk 3 above)
Comment (ytx1991) [reply chia/wallet/did_wallet/did_wallet.py:1105]: It's not bytes32

**Review by Quexington — APPROVED** [review]

**Review by trepca — APPROVED** [review]

**Pull request merged**